### PR TITLE
Update deprecated API's in the documentation.

### LINKED ISSUE #14545: EC_KEY_* is not deprecated in manual pages
Command "grep EC_KEY_ util/libcrypto.num" output only ...DEPRECATEDIN_3_0.... but manual page list only EC_KEY_precompute_mult.

See for instance "https://www.openssl.org/docs/manmaster/man3/EC_KEY_get0_private_key.html"

### PR DESCRIPTION
**Pull request by slontis**
The reported issue related to EC_KEY deprecations
Fixes #14545

Searches were done in the pod files for all libcrypto.num
entries containing DEPRECATEDIN_3_0 to find additional missing entries.

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
@paulidale 
The test failure looks new, but not related to this PR..
````
     # ERROR: (int) 'ossl_property_match_count(q, p) == -1' failed @ test/property_test.c:137
    # [1] compared to [-1]
    # OPENSSL_TEST_RAND_ORDER=1615852722
    not ok 8 - test_property_query_value_create
````

**Comment by paulidale** [comment]
I can't reproduce the failure locally :(

File: doc/man3/EC_GROUP_new.pod [hunk 1]
```diff
@@ -68,6 +60,17 @@ L<openssl_user_macros(7)>:
...
+ int EC_GROUP_set_curve_GFp(EC_GROUP *group, const BIGNUM *p,
+                            const BIGNUM *a, const BIGNUM *b, BN_CTX *ctx);
+ int EC_GROUP_get_curve_GFp(const EC_GROUP *group, BIGNUM *p,
+                            BIGNUM *a, BIGNUM *b, BN_CTX *ctx);
+ int EC_GROUP_set_curve_GF2m(EC_GROUP *group, const BIGNUM *p,
+                             const BIGNUM *a, const BIGNUM *b, BN_CTX *ctx);
+ int EC_GROUP_get_curve_GF2m(const EC_GROUP *group, BIGNUM *p,
+                             BIGNUM *a, BIGNUM *b, BN_CTX *ctx);
+
+
```
Comment (paulidale) [inline doc/man3/EC_GROUP_new.pod:72]: Too many blank lines.

**Review by paulidale — APPROVED** [review]
One nit.

**Comment by slontis** [comment]
> I can't reproduce the failure locally :(

OPENSSL_TEST_RAND_ORDER related??

**Comment by paulidale** [comment]
I tried using the same rng seed without success.

**Comment by mattcaswell** [comment]
Everywhere that we document functions as deprecated we really should be pointing users at the functions they should use instead.

**Comment by slontis** [comment]
Raised issue https://github.com/openssl/openssl/issues/14616 that asks what we should do with deprecated API's that have no equivalents

**Review by paulidale — DISMISSED** [review]

**Comment by slontis** [comment]
@paulidale I am still working on this based on matts comment.

**Comment by slontis** [comment]
Updated deprecation docs with replacement API's and fixed a block copy bug.

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
Merged to master.

**Pull request closed without merging**